src/tcop/tcop.cpp

```diff
@@ -80,35 +81,32 @@ TrafficCop::TcopTxnState &TrafficCop::GetCurrentTxnState() {
   return tcop_txn_state_.top();
 }
 
-ResultType TrafficCop::BeginQueryHelper(const size_t thread_id) {
-  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
-  auto txn = txn_manager.BeginTransaction(thread_id);
-
-  // this shouldn't happen
-  if (txn == nullptr) {
-    LOG_DEBUG("Begin txn failed");
-    return ResultType::FAILURE;
-  }
-
-  // initialize the current result as success
-  tcop_txn_state_.emplace(txn, ResultType::SUCCESS);
-  return ResultType::SUCCESS;
-}
 
 ResultType TrafficCop::CommitQueryHelper() {
+  LOG_TRACE("before commit txn id: %lu",
+           tcop_txn_state_.top().first->GetTransactionId());
   // do nothing if we have no active txns
   if (tcop_txn_state_.empty()) return ResultType::NOOP;
   auto &curr_state = tcop_txn_state_.top();
   tcop_txn_state_.pop();
   // commit the txn only if it has not aborted already
   if (curr_state.second != ResultType::ABORTED) {
+    LOG_TRACE("ENTER != aborted");
     auto txn = curr_state.first;
     auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
     auto result = txn_manager.CommitTransaction(txn);
+
+    // # 623
     return result;
   } else {
     // otherwise, the txn has already been aborted
-    return ResultType::ABORTED;
+    LOG_TRACE("final tcop_txn_state_ size() %lu", tcop_txn_state_.size());
+    auto txn = curr_state.first;
+    auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+    auto result = txn_manager.AbortTransaction(txn);
+    // # 623
```

Comment:
Nobody is going to know what this means when looking at the code. Can you write a better comment?